diff --git a/src/gfx/deinit.c b/src/gfx/deinit.c
index 0560f2e..cba098a 100644
--- a/src/gfx/deinit.c
+++ b/src/gfx/deinit.c
@@ -15,7 +15,7 @@
 
 #include "gfx.h"
 
-#if defined (__linux__)
+#if defined (__linux__) || defined(__EMSCRIPTEN__)
 
 void	gfx_deinit(t_gfx *gfx)
 {
diff --git a/src/hooks/hooks_internal.h b/src/hooks/hooks_internal.h
index 5d3880a..067b3bd 100644
--- a/src/hooks/hooks_internal.h
+++ b/src/hooks/hooks_internal.h
@@ -18,7 +18,7 @@
 
 # include "engine/engine.h"
 
-# if defined(__linux__)
+# if defined(__linux__) || defined(__EMSCRIPTEN__)
 #  include <X11/X.h>
 #  include <X11/keysym.h>
 
diff --git a/src/hooks/init.c b/src/hooks/init.c
index 284d7ec..d7dd208 100644
--- a/src/hooks/init.c
+++ b/src/hooks/init.c
@@ -28,7 +28,7 @@ int	game_destroy_hook(t_engine *e)
 	return (0);
 }
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__EMSCRIPTEN__)
 
 int	hooks_init(t_engine *engine)
 {
